pallets/nfts/src/lib.rs

```diff
@@ -1295,80 +1340,128 @@ pub mod pallet {
-			item: T::ItemId,
+			maybe_item: Option<T::ItemId>,
 			delegate: AccountIdLookupOf<T>,
 		) -> DispatchResult {
-			let maybe_check_origin = T::ForceOrigin::try_origin(origin)
-				.map(|_| None)
-				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
 			let delegate = T::Lookup::lookup(delegate)?;
-			Self::do_cancel_approval(maybe_check_origin, collection, item, delegate)
+			match maybe_item {
+				Some(item) => {
+					let maybe_check_origin =
+						T::ForceOrigin::try_origin(origin).map(|_| None).or_else(|origin| {
+							ensure_signed(origin).map(Some).map_err(DispatchError::from)
+						})?;
+					Self::do_cancel_approval(maybe_check_origin, collection, item, delegate)
+				},
+				None => {
+					let origin = ensure_signed(origin)?;
+					Self::do_cancel_collection_approval(origin, collection, delegate)
+				},
+			}
 		}
 
 		/// Cancel all the approvals of a specific item.
 		///
 		/// Origin must be either:
 		/// - the `Force` origin;
-		/// - `Signed` with the signer being the Owner of the `item`;
+		/// - `Signed` with the signer being the Owner of the `item` if the `item` is provided;
 		///
 		/// Arguments:
 		/// - `collection`: The collection of the item of whose approvals will be cleared.
-		/// - `item`: The item of the collection of whose approvals will be cleared.
+		/// - `maybe_item`: The item of the collection of whose approvals will be cleared. The
+		///   optional item of the collection of whose approval will be cleared. If not provided,
+		///   all approvals to transfer items within collection.
+		/// - `witness_approvals`: Information on the collection approvals cleared. This must be
+		///   correct.
 		///
 		/// Emits `AllApprovalsCancelled` on success.
 		///
 		/// Weight: `O(1)`
 		#[pallet::call_index(17)]
-		#[pallet::weight(T::WeightInfo::clear_all_transfer_approvals())]
+		#[pallet::weight(
+		    T::WeightInfo::clear_all_transfer_approvals(
+				maybe_item.is_some() as u32,
+				T::ApprovalsLimit::get()
+			)
+		)]
 		pub fn clear_all_transfer_approvals(
 			origin: OriginFor<T>,
 			collection: T::CollectionId,
-			item: T::ItemId,
-		) -> DispatchResult {
-			let maybe_check_origin = T::ForceOrigin::try_origin(origin)
-				.map(|_| None)
-				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
-			Self::do_clear_all_transfer_approvals(maybe_check_origin, collection, item)
+			maybe_item: Option<T::ItemId>,
+			witness_approvals: Option<u32>,
+		) -> DispatchResultWithPostInfo {
+			let weight = match maybe_item {
+				Some(item) => {
+					let maybe_check_origin =
+						T::ForceOrigin::try_origin(origin).map(|_| None).or_else(|origin| {
+							ensure_signed(origin).map(Some).map_err(DispatchError::from)
+						})?;
+					Self::do_clear_all_transfer_approvals(maybe_check_origin, collection, item)?;
+					T::WeightInfo::clear_all_transfer_approvals(1, 0)
+				},
+				None => {
+					let origin = ensure_signed(origin)?;
```

Comment:
No force origin consideration.

Reply:
Explained somewhere

Reply:
I do think it should be considered here as this could potentially restrict someone from destroying their collection.  From taking a quick look we would need an upper bound to protect the chain which is possible. However, we need the origin for unreserving the deposit...

@evilrobot-01

Reply:
Origin should be obtainable via iter_prefix but it comes back to bounding. A valid use case, imo, is when a collection is blocked from being removed and governance needs to intervene.